Introducing Excel look and feel and column specified table filtering

Haripriya-Daggubati opened the pull request:
Updated table filtering in such a way that the user feels the Excel type filtering.

In previous scenarios of table filtering, we needed to right-click on the column header to view the search pop-up window. Now we added an icon in the column header for filtering options, and by just clicking on the icon, it shows the search pop-up window.

![Excel_filtering](https://user-images.githubusercontent.com/65753986/230241970-bda40c8e-57c2-4e69-8a85-dcd11a15e88f.png)

And introducing specific column searching along with table searching (search option available for all columns in the table).

![Specific_column_option](https://user-images.githubusercontent.com/65753986/230242162-36591fcf-42b2-443e-a3d5-3fdaf4628cf2.png)

To differentiate filtered columns as part of Excel filtering, introduced two different icons for before and after filtering and also provided the privilege to the user to change the icons as per their needs.

![Before_and_after_filtering_icon](https://user-images.githubusercontent.com/65753986/230243236-a7162292-4cb1-4388-b7b4-11306891a57b.png)

Not as a disadvantage, to more like excel feel removed clear table option in search pop-up window.

prathumca committed:
Added the following

1. Capability to turn on/off sorting
2. Capability to turn on/off filtering for columns

Pending:
Showing menu on filter icon click

prathumca committed:
Added the following

1. Added column icons preferences for filtering
2. Added support to control the right click for filtering
3. When right click is turned off, then clicking on filtering icon only should bring the popup
4. Sorting on/off control

prathumca committed:
1. Fixed creating a Header renderer for each column - now only one renderer will be used to render all columns
2. A Enum is introduced to configure the filter style for fitering columns. Be default, 'DEFAULT' style rendered and all the filtering/filtered icons are not used in default style.

prathumca committed:
Update maven-metadata-local.xml

prathumca committed:
Update maven-metadata-local.xml

prathumca committed:
Update swing-bits-1.0.0-SNAPSHOT.pom

Haripriya-Daggubati committed:
Updated icons and column searching in Excel filtering

1. Updated to display respective column names while searching multiple columns, and Added default filtering and filtered icons
2. Removed clear table icon in popup window

eugener commented on maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/maven-metadata-local.xml:
```diff
@@ -1,35 +1,35 @@
-  <groupId>org.oxbow</groupId>
-  <artifactId>swing-bits</artifactId>
-  <version>1.0.0-SNAPSHOT</version>
+  <groupId>org.oxbow.se</groupId>
```
Group and artifact name should not change

eugener commented on maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/swing-bits-1.0.0-SNAPSHOT.pom:
```diff
@@ -1,9 +1,9 @@
-	<groupId>org.oxbow</groupId>
-	<artifactId>swing-bits</artifactId>
-	<version>1.0.0-SNAPSHOT</version>
+	<groupId>org.oxbow.se</groupId>
```
Group and artifact name should not change

eugener commented on maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/swing-bits-1.0.0-SNAPSHOT.pom:
```diff
@@ -23,7 +23,7 @@
 	<repositories>
 		<repository>
 			<id>oxbow-repository</id>
-			<url>https://github.com/eugener/oxbow/raw/master/maven/repository</url>
```
Please keep the ref to the original repo

eugener commented on maven/repository/org/oxbow/swing-bits/maven-metadata-local.xml:
```diff
@@ -1,13 +1,13 @@
 <metadata>
-  <groupId>org.oxbow</groupId>
-  <artifactId>swing-bits</artifactId>
+  <groupId>org.oxbow.se</groupId>
```
group and artifact name should not change

eugener commented on swingbits/src/main/java/org/oxbow/swingbits/table/filter/TableFilterColumnPopup.java:
```diff
@@ -170,14 +162,15 @@ protected JComponent buildContent() {
-                    return clearAllFilters();
-                }
-            });
+            //removed clear filter for JTable
```
In my opinion, we should not just remove it, but have API to enable/disable it with default being disabled

eugener commented on swingbits/src/main/java/org/oxbow/swingbits/table/filter/TableRowFilterSupport.java:
```diff
@@ -190,13 +204,39 @@ private void setupHeaderRenderers( TableModel newModel, boolean fullSetup ) {
+
+        TableCellRenderer headerRenderer = null;
+        switch (filterType) {
+            case DEFAULT:
```
This is nice - users should be able to choose the renderer

eugener commented:
Thank you for a very nice PR!
I had few comments... Once all the issues are resolved, your code can be integrated.

Haripriya-Daggubati committed:
rollbacked group and artifactId

Haripriya-Daggubati committed:
rolled back as per comment

Haripriya-Daggubati committed:
table clear filter option as API

1. user can enable or disable clear filter in search pop-up window as their requirement
2. rolled back snapshot version

eugener reviewed (approved):
Thank you very much!

Haripriya-Daggubati commented:
Thank you for accepting changes to merge.

I have a request to do from your side. Can we deploy the latest changes to maven repository. So, everyone directly use the jar.

eugener commented:
I will try to make it happen within the next week.

eugener commented:
@Haripriya-Daggubati  

The `v1.3.0` release is available on Maven Central  now 
https://oss.sonatype.org/service/local/repositories/releases/content/org/oxbow/swingbits/

Please let me know if you have any issues.

Haripriya-Daggubati commented:
Thank you so much!

I have no issues.
